tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -1404,6 +1405,65 @@ class Bar(DataClassJsonMixin):
         DataclassTransformer().assert_type(gt, pv)
 
 
+def test_assert_dict_type():
+    @dataclass
+    class Args(DataClassJsonMixin):
+        x: int
+        y: typing.Optional[str]
+
+    # Test when v is a dict
+    vd = {"x": 3, "y": "hello"}
+    DataclassTransformer().assert_type(Args, vd)
+
+    # Test when v is a dict but missing keys from dataclass
+    md = {"x": 3}
+    with pytest.raises(
+        TypeTransformerFailedError,
+        match=re.escape("The original fields are missing the following keys from the dataclass fields: ['y']"),
+    ):
+        DataclassTransformer().assert_type(Args, md)
```

Comment:
`y` is optional, so we don't necessary need to specify it, right?

Reply:
Yes, you're right.